contracts/implementations/ChildGauge.vy

```diff
@@ -52,13 +53,19 @@ event UpdateLiquidityLimit:
     _working_balance: uint256
     _working_supply: uint256
 
+event NewReward:
+    id: indexed(uint256)
+    token: indexed(ERC20)
 
-struct Reward:
+
+struct RewardData:
+    token: ERC20
     distributor: address
-    period_finish: uint256
-    rate: uint256
+    remaining_time: uint256
+    remaining_amount: uint256
     last_update: uint256
     integral: uint256
+    precision: uint256
 
```

Comment:
why is rate not stored here anymore?

I see integral has been taken out, so rate is not needed. I asume this way calculation is done differently then in the past.

I think it would be handy to have the rate stored too, as its more simple to get this over etherscan.

What do you think?

Reply:
Rate is straightly derived from `remaining_amount` and `remaining_time`. I added added corresponding view getters(line 781).

(btw found bug in rate_per_lp, fixed)